simulator/task_manager.go

```diff
@@ -109,3 +179,220 @@ func (m *TaskManager) CreateTask(ctx *Context, req *types.CreateTask) soap.HasFa
 
 	return body
 }
+
+type TaskHistoryCollector struct {
+	mo.TaskHistoryCollector
+
+	*HistoryCollector
+}
+
+func (m *TaskManager) createCollector(ctx *Context, req *types.CreateCollectorForTasks) (*TaskHistoryCollector, *soap.Fault) {
+	if len(m.history.collectors) >= int(m.MaxCollector) {
+		return nil, Fault("Too many task collectors to create", new(types.InvalidState))
+	}
+
+	collector := &TaskHistoryCollector{
+		HistoryCollector: newHistoryCollector(ctx, m.history, defaultPageSize),
+	}
+	collector.Filter = req.Filter
+
+	return collector, nil
+}
+
+// taskFilterChildren returns true if a child of self is the task entity.
+func taskFilterChildren(ctx *Context, root types.ManagedObjectReference, task *types.TaskInfo) bool {
+	seen := false
+
+	var match func(types.ManagedObjectReference)
+
+	match = func(child types.ManagedObjectReference) {
+		if child == *task.Entity {
+			seen = true
+			return
+		}
+
+		walk(ctx.Map.Get(child), match)
+	}
+
+	walk(ctx.Map.Get(root), match)
+
+	return seen
+}
+
+// entityMatches returns true if the spec Entity filter matches the task entity.
+func (c *TaskHistoryCollector) entityMatches(ctx *Context, task *types.TaskInfo, spec *types.TaskFilterSpec) bool {
+	e := spec.Entity
```

Comment:
If `spec` is `nil`, this will NPE. Can `spec` be `nil`?

Reply:
The `spec` can't be nil, changed the param